distributions.py

```diff
@@ -9,6 +9,331 @@ class ProbabilityDistribution:
     def __init__(self):
         pass
 
+class NormalDistribution(ProbabilityDistribution):
+    def __init__(self, mean=None, stdev=None, cov=None, dim=None):
+
+        mean,stdev = _convert_meanstdev_to_iterable(mean, stdev)
+        cov = _convert_cov_to_iterable(cov)
+        
+        self._detect_dimension(mean, stdev, cov, dim)
```

Comment:
I think we can just do `cov` as an input. The input `stdev` is not necessary, and would probably be confusing.

In addition, a standard deviation (`stdev`) is not really well-defined for a multivariate non-isotropic Gaussian. (Is it the Cholesky factor of `cov`? Is it the spd square root of `cov`? Is it the square root of the diagonal of `cov`?